src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/11.diff.sql

```diff
@@ -0,0 +1,150 @@
+
+END
+GO
+
+IF NOT EXISTS 
+(
+    SELECT *
+    FROM    sys.columns
+    WHERE NAME = 'Watermark'
+          AND Object_id = OBJECT_ID('dbo.Workitem')
+)
+BEGIN
+
+    ALTER TABLE dbo.Workitem
+        ADD Watermark BIGINT DEFAULT 0 NOT NULL
+
+END
+GO
+
+/*************************************************************
+    Stored procedure for adding a workitem.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     AddWorkitemV11
+--
+-- DESCRIPTION
+--     Adds a UPS-RS workitem.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The system identifier of the data partition.
+--     @workitemUid
+--         * The workitem UID.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+--     @initialStatus
+--         * New status of the workitem, Either 0(Creating) or 1(Created)
+-- RETURN VALUE
+--     The WorkitemKey
+--     The Watermark
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.AddWorkitemV11
+    @partitionKey                   INT,
+    @workitemUid                    VARCHAR(64),
+    @stringExtendedQueryTags        dbo.InsertStringExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags      dbo.InsertDateTimeExtendedQueryTagTableType_2 READONLY,
+    @personNameExtendedQueryTags    dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY,
+    @initialStatus                  TINYINT
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @watermark BIGINT
+    DECLARE @workitemKey BIGINT
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+
+    SELECT @workitemKey = WorkitemKey
+    FROM dbo.Workitem
+    WHERE PartitionKey = @partitionKey
+        AND WorkitemUid = @workitemUid
+
+    IF @@ROWCOUNT <> 0
+        THROW 50409, 'Workitem already exists', 1;
+
+    -- The workitem does not exist, insert it.
+    SET @workitemKey = NEXT VALUE FOR dbo.WorkitemKeySequence
+    SET @watermark = NEXT VALUE FOR dbo.WorkitemWatermarkSequence
+    INSERT INTO dbo.Workitem
+        (WorkitemKey, PartitionKey, WorkitemUid, Status, Watermark, CreatedDate, LastStatusUpdatedDate)
+    VALUES
+        (@workitemKey, @partitionKey, @workitemUid, @initialStatus, @watermark, @currentDate, @currentDate)
+
+    BEGIN TRY
+
+        EXEC dbo.IIndexWorkitemInstanceCore
+            @partitionKey,
+            @workitemKey,
+            @stringExtendedQueryTags,
+            @dateTimeExtendedQueryTags,
+            @personNameExtendedQueryTags
+
+    END TRY
+    BEGIN CATCH
+
+        THROW
+
+    END CATCH
+
+    COMMIT TRANSACTION
+
+    SELECT
+        @workitemKey,
+        @watermark
+
+END
+GO
+
+COMMIT TRANSACTION
+
+DROP INDEX IF EXISTS IX_Workitem_WorkitemUid_PartitionKey ON dbo.Workitem
+GO
+
```

Comment:
You should first check if the index exists before deleting. But for this we should add DROP_EXISTING = ON